include/seqan3/io/sequence_file/format_sam.hpp

```diff
@@ -0,0 +1,346 @@
+          detail::consume(stream_view | ranges::view::take_while(is_tab));
+          detail::consume(stream_view | ranges::view::take_while(!is_tab));
+          detail::consume(stream_view | ranges::view::take_while(is_tab));
+          detail::consume(stream_view | ranges::view::take_while(!is_tab));
+          detail::consume(stream_view | ranges::view::take_while(is_tab));
+          detail::consume(stream_view | ranges::view::take_while(!is_tab));
+          detail::consume(stream_view | ranges::view::take_while(is_tab));
+
+
+
+          auto seq_view = stream_view | ranges::view::remove_if(is_free || is_digit)   // ignore whitespace and numbers
+                                      | view::take_until_or_throw(is_tab);   // until next space
+
+          // Sequence
+          if constexpr (!detail::decays_to_ignore_v<seq_type>)
+          {
+
+              auto constexpr is_legal_alph = is_in_alphabet<seq_legal_alph_type>;
+              ranges::copy(seq_view | view::transform([is_legal_alph] (char const c) // enforce legal alphabet
+                                      {
+                                          if (!is_legal_alph(c))
+                                          {
+                                              throw parse_error{std::string{"Encountered an unexpected letter: "} +
+                                                                  is_legal_alph.msg.string() +
+                                                                  " evaluated to false on " +
+                                                                  detail::make_printable(c)};
+                                          }
+                                          return c;
+                                      })
+                                    | view::char_to<value_type_t<seq_type>>,      // convert to actual target alphabet
+                           ranges::back_insert_iterator{sequence});
+
+              sequence_size_after = ranges::size(sequence);
+            }
+          else
+          {
+            auto it = ranges::begin(seq_view);
+            auto it_end = ranges::end(seq_view);
+            while (it != it_end)
+            {
+                ++it;
+                ++sequence_size_after;
+            }
+          }
+          detail::consume(stream_view | ranges::view::take_while(is_tab));
+
+          /* Qualities */
+          if constexpr (seq_qual_combined)
+          {
+              // seq_qual field implies that they are the same variable
+              assert(std::addressof(sequence) == std::addressof(qualities));
+              ranges::copy(stream_view | view::char_to<typename value_type_t<qual_type>::quality_alphabet_type>,
+                           ranges::begin(qualities) + sequence_size_before);
+          }
+          else if constexpr (!detail::decays_to_ignore_v<qual_type>)
+          {
+            ranges::copy(stream_view | view::take_until_or_throw(is_cntrl) //TODO: What if we have tags?
```

Comment:
open todo.
> What if we have tags?

then you have a tab instead of a new line so can use `take_until_or_throw(is_cntrl || is_tab)` 

but than make sure to consume the rest.